### PR TITLE
Add comprehensive missing attribution test suite

### PR DESCRIPTION
**Pull request by jwiegley**
## Summary
- 27 new integration tests covering 18 failure hypotheses (H1-H18) for silent AI attribution loss
- Exercises JSONL corruption, SHA mismatches, partial checkpoints, binary files, merge commits, INITIAL file corruption, multi-agent conflicts, and forward-compatibility edge cases
- All tests pass (26 ok, 1 ignored observational)

## Test plan
- [x] `cargo test --test integration missing_attribution -- --test-threads=1` passes (26 ok, 1 ignored)
- [x] `cargo fmt -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/git-ai-project/git-ai/pull/915" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: tests/integration/repos/test_repo.rs [hunk 1]
```diff
@@ -1525,6 +1537,10 @@ impl TestRepo {
                     continue;
                 }
                 if entry.status == "error" {
+                    if self.allow_daemon_errors.load(Ordering::Relaxed) {
+                        completed.insert(session);
+                        continue;
+                    }
```
[the remark this answers is not in the document]
Comment (jwiegley) [reply tests/integration/repos/test_repo.rs:1585]: Fixed in 419b873e. The `allow_daemon_errors` check is now applied in all three daemon-wait functions using Clippy's preferred collapsed `if let ... &&` form.

**Comment by jwiegley** [comment]
Addressed Devin Review's finding about `sanitize_git_env_for_daemon()` and `disable_trace2_for_daemon_process()` being called with tokio worker threads already alive (UB on POSIX).

Fixed in 7a37d840: moved both calls from `run_daemon()` (inside `runtime.block_on()`) to `handle_run()` before `Builder::new_multi_thread().build()`, ensuring no other threads exist when `env::remove_var`/`env::set_var` execute.

File: src/daemon.rs [hunk 2]
```diff
@@ -7192,8 +7193,6 @@ pub(crate) fn daemon_run_pending_self_update() {
 }
 
 pub async fn run_daemon(config: DaemonConfig) -> Result<(), GitAiError> {
-    sanitize_git_env_for_daemon();
-    disable_trace2_for_daemon_process();
     config.ensure_parent_dirs()?;
```
[the remark this answers is not in the document]
Comment (jwiegley) [reply src/daemon.rs]: Fixed in e9f51d56 — `run_daemon` is now `pub(crate)` since it's only called from `handle_run`. This ensures external callers can't bypass the env-sanitization prerequisites.

**Comment by jwiegley** [comment]
* **#915** <a href="https://app.graphite.com/github/pr/git-ai-project/git-ai/915?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/git-ai-project/git-ai/915?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
